upgrade_common_services.sh

```diff
@@ -0,0 +1,388 @@
+#!/bin/bash
+#
+# Copyright 2021 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# counter to keep track of installation steps
+STEP=0
+
+# script base directory
+BASE_DIR=$(dirname "$0")
+
+CS_LIST_CSNS=("operand-deployment-lifecycle-manager-app"
+        "ibm-common-service-operator"
+        "ibm-cert-manager-operator"
+        "ibm-mongodb-operator"
+        "ibm-iam-operator"
+        "ibm-monitoring-grafana-operator"
+        "ibm-healthcheck-operator"
+        "ibm-management-ingress-operator"
+        "ibm-licensing-operator"
+        "ibm-commonui-operator"
+        "ibm-ingress-nginx-operator"
+        "ibm-auditlogging-operator"
+        "ibm-platform-api-operator"
+        "ibm-namespace-scope-operator"
+        "ibm-namespace-scope-operator-restricted"
+        "ibm-zen-operator"
+        "ibm-zen-cpp-operator"
+        "ibm-crossplane-operator-app"
+        "ibm-crossplane-provider-ibm-cloud-operator-app"
+        "ibm-crossplane-provider-kubernetes-operator-app")
+
+CS_LIST_CONTROLNS=(
+        "ibm-cert-manager-operator"
+        "ibm-namespace-scope-operator"
+        "ibm-namespace-scope-operator-restricted"
+        "ibm-crossplane-operator-app"
+        "ibm-crossplane-provider-ibm-cloud-operator-app"
+        "ibm-crossplane-provider-kubernetes-operator-app")
+
+# ---------- Command functions ----------
+# ---------- Command functions ----------
+function usage() {
+	local script="${0##*/}"
+
+	while read -r ; do echo "${REPLY}" ; done <<-EOF
+	Usage: ${script} [OPTION]...
+	Upgrade Common Services
+
+	Options:
+	Mandatory arguments to long options are mandatory for short options too.
+      -h, --help                    display this help and exit
+      -csNS                         specify the namespace where common service is installed. By default it is namespace ibm-common-services.
+      -cloudpaksNS                  specify the namespace where cloud paks is installed. By default it would be same as csNS.
+      -controlNS                    specify the namespace where singleton services are installed. By default it it would be same as csNS.
+      -c                            specify the subscription channel where common services switch. By default it is channel v3
+	Note:
+	If there is no namespace defined through arguments, all Common Service in the cluster will be upgraded.
```

Comment:
I don't think this has to be changed now, but if we were to continue with the script approach down the line, I could see this causing some headaches in multi instance clusters where specific versions of cs are installed

Reply:
It should be changed to by default, the namespace you are currently one will be selected instead of all namespaces. All namespaces should be very explicit.

Reply:
Will fix in another PR